src/Coduo/PHPHumanizer/Number/Spell.php

```diff
@@ -0,0 +1,148 @@
+class Spell
+{
+    /**
+     * @var array
+     */
+    private $map = array(
+        0                   => 'zero',
+        1                   => 'one',
+        2                   => 'two',
+        3                   => 'three',
+        4                   => 'four',
+        5                   => 'five',
+        6                   => 'six',
+        7                   => 'seven',
+        8                   => 'eight',
+        9                   => 'nine',
+        10                  => 'ten',
+        11                  => 'eleven',
+        12                  => 'twelve',
+        13                  => 'thirteen',
+        14                  => 'fourteen',
+        15                  => 'fifteen',
+        16                  => 'sixteen',
+        17                  => 'seventeen',
+        18                  => 'eighteen',
+        19                  => 'nineteen',
+        20                  => 'twenty',
+        30                  => 'thirty',
+        40                  => 'fourty',
+        50                  => 'fifty',
+        60                  => 'sixty',
+        70                  => 'seventy',
+        80                  => 'eighty',
+        90                  => 'ninety',
+        100                 => 'hundred',
+        1000                => 'thousand',
+        1000000             => 'million',
+        1000000000          => 'billion',
+        1000000000000       => 'trillion',
+        1000000000000000    => 'quadrillion',
+        1000000000000000000 => 'quintillion'
+    );
+
+    /**
+     * @param $number
+     * @param $locale
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function __construct($number, $locale = 'en')
+    {
+        if (!is_numeric($number)) {
+            throw new \InvalidArgumentException("Spell number converter only accepts numeric values.");
+        }
+
+        if (($number >= 0 && (int) $number < 0) || (int) $number < 0 - PHP_INT_MAX) {
+            throw new \InvalidArgumentException("Spell number converter only accepts numeric values between -" . PHP_INT_MAX . " and " . PHP_INT_MAX . ".");
+        }
+
+        $this->number      = $number;
+        $this->locale      = $locale;
+        $this->translator  = Builder::build($this->locale);
+        $this->hyphen      = $this->translate('hyphen');
+        $this->conjunction = ' ' . $this->translate('conjunction') . ' ';
+        $this->separator   = $this->translate('separator') . ' ';
+        $this->negative    = $this->translate('negative') . ' ';
+        $this->decimal     = ' ' . $this->translate('decimal') . ' ';
+    }
+
+    /**
+     * @return string
+     */
+    public function convert($number = null)
+    {
+        if (is_null($number)) {
+            $number = $this->number;
+        }
+
+        if ($number < 0) {
+            return $this->negative . $this->convert(abs($number));
```

Comment:
Now when you removed ``private $negative`` you should change this line into something like that:
```
return $this->translate('negative') . ' ' . $this->convert(abs($number));
```

(same in other cases)